src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -50,28 +60,177 @@
+        .filter(isWeekend.negate())
+        .collect(Collectors.toList());
+
+    // Convert to list of strings in proper format
+    List<String> dateList = new ArrayList<>();
+    for (LocalDate lD : businessDays) {
+      dateList.add(lD.format(DateTimeFormatter.ofPattern("yyyyMMdd")));
+    }
+    return dateList;
+  }
+
+  /**
+   * Chooses whether to get the historical price for the symbol from the repository or Iex.
+   * Precedence is always given to the repository, unless is not found within it.
+   *
+   * @param symbol Stock symbol to get the historical price for.
+   * @param date Input date (formatted YYYYMMDD) to get price data for.
+   *      Note: We will convert this into YYYY-MM-DD format for repository search.
+   *      - Format of Iex API output
+   * @return a list (length = 1) of IexHistoricalPrices objects.
+   */
+  public List<IexHistoricalPrices> getHistoricalPriceBySymbolAndDate(
+      final String symbol, final String date) {
+    //First, try the database
+    List<IexHistoricalPrices> dbList = new ArrayList<>();
+    List<IexHistoricalPricesDB> entry =
+        repository.findBySymbolAndDate(symbol, convertDateFormatToOutput(date));
+
+    if (!entry.isEmpty()) {
+      dbList.add(new IexHistoricalPrices(entry.get(0)));
+      return dbList;
+    } else {
+      //If not in database, call from Iex, insert into database, and return
+      List<IexHistoricalPrices> call = iexClient.getHistoricalPricesForSymbolByDate(symbol, date);
+      if (!call.isEmpty()) {
+        repository.save(new IexHistoricalPricesDB(call.get(0)));
+      }
+      return call;
+    }
+  }
+
+  /**
+   * Converts input date (format:YYYYMMDD) to output date (format:YYYY-MM-DD).
+   * @param inputDate date string in YYYYMMDD format
+   * @return outputDate string in YYYY-MM-DD format.
+   */
+  public String convertDateFormatToOutput(final String inputDate) {
+    StringBuilder builder = new StringBuilder(inputDate);
+    builder.insert(6,'-');
+    builder.insert(4,'-');
+    return builder.toString();
+  }
+
+  /**
+   * Calls Iex API client to obtain historical prices for full query.
+   *
+   * @param symbol list of symbols to get historical price for.
+   * @param range the range of time  (ex. "5m", "ytd" ) to get previous data (Optional).
+   * @param date the date from which we would want to get the previous data from (Optional).
+   *      Note: If neither optional parameter is used, the system will default to range = 1m.
+   *      - default behavior of Iex Cloud API
+   *      Note: If both optional parameters are provided, the system will default to the date.
+   *      - can be changed later if needed: IexCloud will not allow both options in call to client
+   * @return a list of IexHistoricalPrices objects for the given symbols.
+   */
+  public List<IexHistoricalPrices> getHistoricalPricesFromIex(
+      final String symbol, final String range, final String date) {
+    List<IexHistoricalPrices> call;
```

Comment:
This variable name is not descriptive.

Reply:
Changed variable name to pricesFromIexApi.